### ISSUE TITLE
Incorrect error reported when using pandas dataframes

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#593]
```
import pandas as pd
df = pd.DataFrame([0, 1, 3])
df2 = df.iloc[:, [0, 1]]
row = df.loc[df.index[0], :]
```

**Comment by erictraut** [comment]
The message is:
```
Object of type 'NoReturn' does not define '__getitem__'
```

I've investigated the problem, and the type checker is reporting the proper error, even if it's confusing. The problem is that DataFrame's base class (NDFrame) defines a method `__getattr__` that calls `__getitem__` which calls `self._get_item_cache`. This method calls `self._box_item_values`, which always raises an exception. If this method were marked abstract (using the built-in decorator), pyright would do the right thing and assume that the method would be overridden, but it's not marked as such. 

The workarounds to this problem include:
1. Type errors should not be reported for users who are not interested in them.
2. Users who are interested in type errors should find and use a high-quality type stub library for pandas.
3. Someone should contribute a change to pandas to mark `_box_item_values` as abstract.
